server/src-lib/Hasura/Server/Init.hs

```diff
@@ -1010,3 +1023,15 @@ downgradeOptionsParser =
         ( long ("to-" <> v) <>
           help ("Downgrade to graphql-engine version " <> v <> " (equivalent to --to-catalog-version " <> catalogVersion <> ")")
         )
+
+connectionCompressionEnv :: (String, String)
+connectionCompressionEnv =
+  ( "HASURA_GRAPHQL_CONNECTION_COMPRESSION"
+  , "Enable Websocket Compression (default: false)"
+  )
+
+parseConnectionCompression :: Parser Bool
+parseConnectionCompression =
+  switch ( long "connection-compression" <>
+           help (snd connectionCompressionEnv)
+         )
```

Comment:
I think this PR generally looks great. My only quibble is that I think “connection compression” is pretty vague. I think it would be helpful to be a little more specific:

```suggestion
webSocketCompressionEnv :: (String, String)
webSocketCompressionEnv =
  ( "HASURA_GRAPHQL_WEBSOCKET_COMPRESSION"
  , "Enable WebSocket permessage-deflate compression (default: false)"
  )

parseWebSocketCompression :: Parser Bool
parseWebSocketCompression =
  switch ( long "websocket-compression" <>
           help (snd webSocketCompressionEnv)
         )
```

This way, we’re at least a little more clear about what the compression scheme is, and we’ll have hedged our bets in case some other compression scheme becomes popular in the future.

(Of course, the use sites of these bindings would have to be updated, too, but that would be awkward to do as a GH PR suggestion.)

Reply:
Fair point! I'll adjust it.